File: python_backend/restriction/restriction.py
from custom_except import *
from helpers.all_xy import all_xy


def restriction(coord_list):
    """get a restriction set from the coord_list (threat_area or pins)
    if coord_list was empty, then no restriction was found so return set of all__xy else convert coord_list to
    a set and return
    """
    if not coord_list:
        return all_xy
    else:
        return set(coord_list)

File: app (1)/src/game_logic/ranges/restriction.js
import {allXy} from "gameLogic/misc/allXy";


export function restriction(coordList) {
    /**get a restriction set from the coordList (threatArea || pins)
    if coordList was empty, then no restriction was found so return set of all_Xy else convert coordList to
    a set &&  return
    */
    if (! coordList) {
        return allXy
    }
    else {
        return set(coordList)
    }
}